project/SparkBuild.scala

```diff
@@ -257,7 +257,7 @@ object SparkBuild extends PomBuild {
 
   lazy val sharedSettings = sparkGenjavadocSettings ++
                             compilerWarningSettings ++
-      (if (sys.env.contains("NOLINT_ON_COMPILE")) Nil else enableScalaStyle) ++ Seq(
+      (if (!sys.env.get("NOLINT_ON_COMPILE").contains("false")) enableScalaStyle else Nil) ++ Seq(
```

Comment:
1. no env variable => `enableScalaStyle`
2. `NOLINT_ON_COMPILE=1` => `enableScalaStyle`
3. `NOLINT_ON_COMPILE=true` => `enableScalaStyle`
4. `NOLINT_ON_COMPILE=false` => Nil

Reply:
nit: Shall we rename to `LINT_ON_COMPILE` and only disables it when `NOLINT_ON_COMPILE=false`?  Seems easier to understand

Reply:
Ya, I thought in that way, but it's required for backward compatibility.

Reply:
The existing semantic has been used since SPARK-14790 for 8 years.

Reply:
ok, I don't have a strong opinion here.